research/gaia-x/pegasus/pegasus/src/operator/concise/limit.rs

```diff
@@ -53,3 +60,213 @@ impl<D: Data> Limit<D> for Stream<D> {
         })
     }
 }
+
+struct SmallHeap<D: Ord> {
```

Comment:
`FixedSizeHeap` may be a better name.

Reply:
Can we combine CustomHeap as well, do something like:
```
struct FixedSizeHeap<D> {
    pub limit: usize,
    pub heap: BinaryHeap<D>,
    cmp: Rc<Fn(&D, &D) -> Ordering + Send>
}

impl<D: Ord> FixedSizeHeap {
    pub fn new(limit: usize) -> Self {
        ...
       Self {
           limit,
           heap,
           cmp: Rc::new(|x: &D, y: &D| x.cmp(y))
       }
    } 
}

impl<D> FixedSizeHeap {
    pub fn with_cmp(limit: usize, cmp: Rc<...>) -> Self {
        ...
       Self {
           limit,
           heap,
           cmp
       }
    } 
}


```

Reply:
fixed